src/core/tiledscene/qgsesrii3sdataprovider.cpp

```diff
@@ -811,31 +853,45 @@ bool QgsEsriI3SDataProvider::loadFromSlpk( const QString &uri, json &layerJson,
   }
   catch ( const json::parse_error & )
   {
-    appendError( QgsErrorMessage( tr( "Unable to parse metadata JSON: " ) + uri, QStringLiteral( "I3S" ) ) );
+    appendError( QgsErrorMessage( tr( "Unable to parse metadata JSON: " ) + metadataFileName, QStringLiteral( "I3S" ) ) );
```

Comment:
why this change to `metadataFileName` ? that's just "metadata.json" so it does not tell user much - full path would be more useful I think...

(also below)